src/@next/utils/core.ts

```diff
@@ -36,3 +39,13 @@ export const generatePageUrl = (slug: string) => `/page/${slug}/`;
 
 export const generateGuestOrderDetailsUrl = (token: string) =>
   `/order-history/${token}/`;
+
+export const checkIfShippingRequiredForProducts = (items?: IItems) =>
+  items?.some(({ variant }) => variant.product?.productType.isShippingRequired);
+
+export const checkIfCheckoutPriceEqualPaymentPrice = (
```

Comment:
If you want to test it you have to mock the whole payment object and total price. I see 2 solutions:
1. Prefereble, easier to implement: Don't we have util for comparing prices? 
If not, I would suggest creating utils/money.ts file with such function (having all utils in one file leads to one, too long file with too broad scope - nightmare to work with).
Also, unit tests for such function is a must. 
2. Function `isCheckoutFullyPaid` which takes Checkout as an argument and performs needed checks. Thats more to mock, but could be easier to use in checkout views.

Reply:
I have created isPriceEqual util

Reply:
Another way to do it could be to update the available shipping methods when the cart is updated, then check the id of the available shipping methods against the id of the chosen shipping method in the checkout router.